### PR TITLE
HMS-3162: test: boot generated image and ensure ssh is working

### PR DESCRIPTION
**Pull request by mvo5**
This PR adds the extra check that the generated image from `podman run osbuild-deploy-container` can actually boot in qemu and will open the ssh port.

### CODE REVIEW

File: plans/all.fmf [hunk 1]
```diff
@@ -2,6 +2,9 @@ summary: Run all tests inside a VM environment
 provision:
   how: virtual
   image: fedora:39
+  hardware:
+    virtualisation:
```
Comment (thrix) [inline plans/all.fmf:6]: https://tmt.readthedocs.io/en/stable/spec/hardware.html#virtualization

```suggestion
    virtualization:
```

**Review by thrix — COMMENTED** [review]
typo here

File: plans/all.fmf (hunk 1 above)
Comment (mvo5) [reply plans/all.fmf]: Thanks! Fixed

**Comment by thrix** [comment]
@mvo5 looks the action is not yet working well with RH ranch :(

It fails to report the correct URL seems.
That would be:

http://artifacts.osci.redhat.com/testing-farm/54e53734-4050-45df-abff-c79e365d354c/

So seems tests run, but they failed on:

```
E               FileNotFoundError: [Errno 2] No such file or directory: 'qemu-system-x86_64'
```

**Comment by thrix** [comment]
@mvo5 never mind, you need to adjust `tf_scope` to `private`:

https://github.com/sclorg/testing-farm-as-github-action/blob/main/action.yml#L78

File: test/test_smoke.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    # check that there are no denials
+    # TODO: actually check this once https://github.com/osbuild/images/pull/287
+    #       is merged
+    journal_output = testutil.journal_after_cursor(cursor)
+    assert journal_output != ""
+    generated_img = pathlib.Path(output_path) / "qcow2/disk.qcow2"
+    assert generated_img.exists(), f"output file missing, dir content: {os.listdir(os.fspath(output_path))}"
+    with VM(generated_img) as test_vm:
+        test_vm.start()
+        # TODO: replace with 'test_vm.run("true")' once user creation via
```
[the remark this answers is not in the document]
Comment (mvo5) [reply test/test_smoke.py:68]: I prepared a followup for this in https://github.com/osbuild/bootc-image-builder/compare/main...mvo5:bootc-image-builder:origin/tmt-smoke-boot-vm-2?expand=1

**Comment by achilleas-k** [comment]
Needs a rebase and conflict resolution.

**Review by ondrejbudai — APPROVED** [review]

**Pull request merged**